crypto/cms/cms_pwri.c

```diff
@@ -201,14 +201,15 @@ static int kek_unwrap_key(unsigned char *out, size_t *outlen,
                           const unsigned char *in, size_t inlen,
                           EVP_CIPHER_CTX *ctx)
 {
-    size_t blocklen = EVP_CIPHER_CTX_get_block_size(ctx);
+    int blocksz = EVP_CIPHER_CTX_get_block_size(ctx);
+    size_t blocklen = blocksz;
     unsigned char *tmp;
     int outl, rv = 0;
     if (inlen < 2 * blocklen) {
         /* too small */
         return 0;
     }
-    if (inlen % blocklen) {
+    if (blocksz <= 0 || inlen % blocklen) {
         /* Invalid size */
         return 0;
     }
```

Comment:
I would also put this if before the previous one. Funnily enough this newly added condition actually will never be evaluated for blocksz being negative as the blocklen value in the previous `if` will be a huge number and thus the condition `inlen < 2 * blocklen` will be true.